maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java

```diff
@@ -121,11 +121,11 @@ public CLIManager()
         options.addOption( Option.builder( Character.toString( QUIET ) ).longOpt( "quiet" ).desc( "Quiet output - only show errors" ).build() );
         options.addOption( Option.builder( Character.toString( DEBUG ) ).longOpt( "debug" ).desc( "Produce execution debug output" ).build() );
         options.addOption( Option.builder( Character.toString( ERRORS ) ).longOpt( "errors" ).desc( "Produce execution error messages" ).build() );
-        options.addOption( Option.builder( Character.toString( NON_RECURSIVE ) ).longOpt( "non-recursive" ).desc( "Do not recurse into sub-projects" ).build() );
+        options.addOption( Option.builder( Character.toString( NON_RECURSIVE ) ).longOpt( "non-recursive" ).desc( "Do not recurse into sub-projects. When used together with -pl, do not recurse into sub-projects of selected aggregators" ).build() );
         options.addOption( Option.builder( Character.toString( UPDATE_SNAPSHOTS ) ).longOpt( "update-snapshots" ).desc( "Forces a check for missing releases and updated snapshots on remote repositories" ).build() );
-        options.addOption( Option.builder( Character.toString( ACTIVATE_PROFILES ) ).longOpt( "activate-profiles" ).desc( "Comma-delimited list of profiles to activate" ).hasArg().build() );
+        options.addOption( Option.builder( Character.toString( ACTIVATE_PROFILES ) ).longOpt( "activate-profiles" ).desc( "Comma-delimited list of profiles to activate. Prefixing a profile with ! excludes it, and ? marks it as optional" ).hasArg().build() );
```

Comment:
I would rather write: ...with an exclamation mark (!) ex...., and question mark (?) marks it...

Reply:
We strived for conciseness, as this text is now already four lines long (for profiles already seven, which makes it the longest option in mvn help by far). By writing out the symbols in full, this will increase even further. Are you sure you want this?

Reply:
This is my concern too. The (default) help output should be compact, it used to fit on 1 page. 
With the new jansi we're probably gaining some space due to the width calculation.

Reply:
Ah, that sounds good. When there's more space, we should take a look at explaining the flags a bit better I think. But perhaps for now we should stick with this?

Reply:
Let's not await that Jansi upgrade, it's [been open for quite some time](https://github.com/apache/maven/pull/433) already. If/when we have more screen real estate, we can revisit the whole `mvn -h` output.

Reply:
OK guys, let's add as-is and re-review as a whole later.